pkg/tfbridge/provider.go

```diff
@@ -163,6 +163,15 @@ type DataSource struct {
 	TFName string          // the Terraform resource name.
 }
 
+type CheckFailureError struct {
```

Comment:
Doc comment here referencing PreConfigureCallback, and a doc comment on PreConfigureCallback noting that the user code may return CheckFailureError perhaps?